pkg/activator/handler/handler.go

```diff
@@ -61,7 +72,45 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	err := a.Throttler.Try(revID, func() {
-		attempts, httpStatus := a.proxyRequest(w, r, target)
+		var attempts int
+		var httpStatus int
+		// If a GET probe interval has been configured, then probe
+		// the queue-proxy with our network probe header until it
+		// returns a 200 status code.
+		success := (a.GetProbeCount == 0)
+		probeReq := &http.Request{
+			Method:     http.MethodGet,
+			URL:        target,
+			Proto:      r.Proto,
+			ProtoMajor: r.ProtoMajor,
+			ProtoMinor: r.ProtoMinor,
+			Host:       r.Host,
+			Header: map[string][]string{
+				http.CanonicalHeaderKey(network.ProbeHeaderName): []string{"true"},
+			},
+		}
+		for attempts = 0; attempts < a.GetProbeCount && !success; attempts++ {
+			probeResp, err := a.Transport.RoundTrip(probeReq)
+			if err != nil {
+				a.Logger.Errorw("Pod probe failed", zap.Error(err))
+				time.Sleep(getProbeInterval)
+				continue
+			}
+			if probeResp.StatusCode != http.StatusOK {
+				httpStatus = probeResp.StatusCode
+				a.Logger.Errorf("Pod probe sent status: %d", probeResp.StatusCode)
```

Comment:
Does it occur too often? 
Just thinking whether we need to print out the number of retries rather then flooding the activator with many similar lines.

Reply:
I would expect 1-3 per cold-start, so not huge.